app/presenters/generic_phrases_presenter.rb

```diff
@@ -0,0 +1,93 @@
+class GenericPhrasesPresenter
+  include Rails.application.routes.url_helpers
+  attr_reader :pagination, :sorting, :search_params, :url_params, :options, :items, :verbs, :adjectives, :verb, :adjective
+  delegate :page, :total_pages, :total_items, to: :pagination
+  delegate :sort_key, :sort_direction, to: :sorting
+
+  def initialize(generic_phrases, verbs, adjectives, url_params, options)
+    @search_params = search_params
+    @pagination = PaginationPresenter.new(page: options[:page], total_items: generic_phrases.count)
+    @sorting = SortPresenter.new(sort_key: options[:sort_key], sort_direction: options[:sort_direction])
+    @url_params = url_params
+    @options = options
+
+    @items = pagination.paginate(generic_phrases)
+    @verbs = verbs
+    @adjectives = adjectives
+
+    @verb = options[:verb]
+    @adjective = options[:adjective]
+  end
+
+  def table_head
+    [
+      generic_phrase_head,
+      verb_head,
+      topic_head,
+    ]
+  end
+
+private
+
+  def generic_phrase_head
+    key = "generic_phrase"
+    {
+      text: "Generic phrase",
+      href: href(key),
+    }.merge(sort_direction(key))
+  end
+
+  def verb_head
+    key = "verb"
+    {
+      text: "Verb",
+      href: href(key),
+    }.merge(sort_direction(key))
+  end
+
+  def topic_head
+    key = "adj"
+    {
+      text: "Topic",
+      href: href(key),
+    }.merge(sort_direction(key))
+  end
+
+  def href(key)
+    direction_for_link = "asc"
+    if currently_sorting_by_key?(key)
+      direction_for_link = opposite_sort_dir
+    end
+
+    sort_params = { sort_key: key, sort_direction: direction_for_link }
+    generic_phrases_path(url_params.merge(sort_params).merge(filter_params))
+  end
+
+  def filter_params
+    {
+      verb: options[:verb],
+      adjective: options[:adjective],
+    }
+  end
+
+  def currently_sorting_by_key?(key)
+    key == sort_key
+  end
+
+  def opposite_sort_dir
```

Comment:
I'm sure we're on the same wavelength here so more of a note for the future, this wants to be moved to somewhere it can be shared with the PagesPresenter

Reply:
Absolutely. Sorting needs revisiting as a whole.